charts/synthetics-private-location/templates/deployment.yaml

```diff
@@ -61,4 +61,4 @@ spec:
       volumes:
       - name: worker-config
         configMap:
-          name: {{ include "synthetics-private-location.fullname" . }}-config
+          name: {{ include "synthetics-private-location.configMapName" . }}
```

Comment:
I think the volume should to be added only if `.Values.configMap.enabled`

Reply:
In my use case, I need the volume added but will be creating it out of band.

Reply:
I could add an extra volumes/volumeMounts section to the chart and only conditionally add the configMap that was created when enabled?

Reply:
I think we can simplify the logic by removing `configMap.enabled` parameter and test `{{- if .Values.configMap.name == "" }}` to create or configmap in the file `charts/synthetics-private-location/templates/configmap.yaml`

WDYT ?

Reply:
👍 Let me make that change

Reply:
Thinking about this more, I think we should keep the `enabled` parameter. I see the following 2 use cases:
1. As currently written where the configMap is created and managed by the Helm chart
2. ConfigMap is managed out of band and used by the synthetics.

For option 1, we probably don't want to specify the name (allow the defaults). For option 2, we may want to specify the name or use the default. Based off this, I'd say the name only matters when providing your own config map so we shouldn't use it for the default case.

Reply:
I agree with the 2 use cases that you mention but:
1. the current generated configMap name is generated with the release name.
2. If someone provide `.values.configMap.name` it means that the ConfigMap is managed outside the chart. So we dont really need to have the `enabled` field.

Maybe the issue is only how we name this new field. WDYT if we renamed it: `externalConfiguration.configMap.name`

```yaml
externalConfiguration:
  configMap:
    # externalConfiguration.configMap.name -- ConfigMap name containing `synthetics-check-runner.json` configuration file. if set, it overrides the `configFile` parameter.
    name: ""
```

Reply:
That makes it clear it's external. I'll make that change